common/djangoapps/student/models.py

```diff
@@ -1590,7 +1591,12 @@ def enrollments_for_user(cls, user):
         return cls.objects.filter(user=user, is_active=1).select_related('user')
 
     @classmethod
-    def enrollments_for_user_with_overviews_preload(cls, user):  # pylint: disable=invalid-name
+    def get_dashboard_course_limit(cls):
+        course_limit = getattr(settings, 'DASHBOARD_COURSE_LIMIT')
+        return course_limit
+
+    @classmethod
+    def enrollments_for_user_with_overviews_preload(cls, user, load_all_courses=False):  # pylint: disable=invalid-name
```

Comment:
The same thing I wrote last time about backwards compatibility applies here. When you're adding a default argument, you should try to make it match the old behavior if possible. In this case, it would be setting `load_all_courses` to be `True`. In this case, this method is also used in `contact_us.py` and may introduce a bug where the support team gets a request from this user for a course that doesn't show up in their enrollments (because that was truncated off).

Now normally, you could address that fine by changing the two or three places in edx-platform that actually use this, but this method is also likely used by people extending the platform, since it's a really common thing to want to get the enrollments for a user. So please alter the method to return exactly what it would have by default if the optional arg isn't sent.

Also, you might want to consider making the optional argument be an integer for how many courses to load instead of a boolean. The student dashboard view might want that to be `settings.DASHBOARD_COURSE_LIMIT` number of courses, but that setting doesn't have to reach down into the model level -- it's plausible that other people calling it would want to use different numbers in different situations. You'd still use your setting, but you could pass it in from the view instead of accessing the setting from inside the model code.